File: NoroffAlumni/src/main/java/no/experisAcadmey/trondheim/NoroffAlumni/models/DTOs/postDTOs/PostDto.java
package no.experisAcadmey.trondheim.NoroffAlumni.models.DTOs.postDTOs;

import lombok.Data;
import no.experisAcadmey.trondheim.NoroffAlumni.models.DTOs.groupDTOs.GroupReducedDto;
import no.experisAcadmey.trondheim.NoroffAlumni.models.DTOs.replyDTOs.ReplyDto;
import no.experisAcadmey.trondheim.NoroffAlumni.models.DTOs.topicDTOs.TopicReducedDto;
import no.experisAcadmey.trondheim.NoroffAlumni.models.DTOs.userDTOs.UserDisplayDto;
import no.experisAcadmey.trondheim.NoroffAlumni.models.DTOs.userDTOs.UserDto;

import java.util.Date;
import java.util.List;

@Data
public class PostDto {
    private Long id;
    private String title;
    private String body;
    private Date last_updated;
    private UserDisplayDto author;
    private UserDisplayDto target_user;
    private String authorInitlas;
    
    private List<TopicReducedDto> target_topics;
    private List<GroupReducedDto> target_group;

    private List<ReplyDto> replies;
}

File: noroff-alumni-frontend/src/models/PostDTO.ts
import UserDisplayDTO from "./UserDisplayDTO";
import ReplyDTO from "./ReplyDTO";
import TopicReducedDTO from "./TopicReducedDTO";
import GroupReducedDTO from "./Group/GroupReducedDTO";

type PostDTO = {
    id: number,
    title: string,
    body: string,
    last_updated: Date,
    author: UserDisplayDTO,
    target_user?: UserDisplayDTO,
    target_topics?: TopicReducedDTO[],
    target_group?: GroupReducedDTO[],
    replies?: ReplyDTO[];
}

export default PostDTO;
